src/data/monoid_algebra.lean

```diff
@@ -679,6 +684,42 @@ lemma alg_hom_ext_iff {R : Type u₃} [comm_semiring k] [add_monoid G]
   (∀ x, φ₁ (finsupp.single x 1) = φ₂ (finsupp.single x 1)) ↔ φ₁ = φ₂ :=
 ⟨λ h, alg_hom_ext h, by rintro rfl _; refl⟩
 
+/--
+The `alg_hom` which maps from a grading of an algebra `A` back to that algebra.
```

Comment:
I wouldn't use the word "grading" in this context. Rather, I would call this map the augmentation map of the monoid algebra: https://ncatlab.org/nlab/show/augmentation+ideal#for_group_algebras

For the Lean identifier, either `augmentation` or something like your `sum_id` or perhaps `sum_coeffs` seems fine.

Reply:
Would you mind suggesting a complete docstring? My mathematical background is severely lacking.